src/helpers/pivot/spreadsheet_pivot/spreadsheet_pivot.ts

```diff
@@ -0,0 +1,384 @@
+      return false;
+    }
+    for (const measure of this.definition.measures) {
+      if (!measure.isValid) {
+        return false;
+      }
+    }
+    for (const dimension of this.definition.columns.concat(this.definition.rows)) {
+      if (!dimension.isValid) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  assertIsValid({ throwOnError }: { throwOnError: boolean }): FPayload | undefined {
+    if (!this.isValid()) {
+      if (throwOnError) {
+        if (this.invalidRangeError) {
+          throw this.invalidRangeError;
+        } else {
+          throw new EvaluationError(_t("At least one measure and/or dimension is not correct."));
+        }
+      }
+      return {
+        value: CellErrorType.GenericError,
+        message:
+          this.invalidRangeError?.message ??
+          _t("At least one measure and/or dimension is not correct."),
+      };
+    }
+    return undefined;
+  }
+
+  markAsDirtyForEvaluation(): void {
+    this.isDirtyForEvaluation = true;
+  }
+
+  getMeasure(name: string): PivotMeasure {
+    return this.definition.getMeasure(name);
+  }
+
+  getPivotHeaderValueAndFormat(domainStr: StringDomainArgs): FPayload {
+    const domain = toDomainArgs(domainStr);
+    const lastNode = domain.at(-1);
+    if (!lastNode) {
+      return { value: _t("Total") };
+    }
+    if (lastNode.field === "measure") {
+      return { value: this.getMeasure(lastNode.value).displayName };
+    }
+    const dimension = this.getDimension(lastNode.field);
+    const cells = this.filterDataEntriesFromDomain(this.dataEntries, domain);
+    const finalCell = cells[0]?.[dimension.nameWithGranularity];
+
+    if (!finalCell) {
+      return { value: "" };
+    }
+    if (finalCell.value === null) {
+      return { value: _t("(Undefined)") };
+    }
+    if (dimension.type === "date") {
+      if (dimension.granularity === "day") {
+        return {
+          value: toNumber(finalCell.value, this.getters.getLocale()),
+          format: this.getters.getLocale().dateFormat,
+        };
+      }
+      if (dimension.granularity === "month_number") {
+        return {
+          value: MONTHS[toNumber(finalCell.value, this.getters.getLocale())].toString(),
+        };
+      }
+    }
+    return {
+      value: finalCell.value,
+      format: finalCell.format,
+    };
+  }
+
+  getPivotCellValueAndFormat(measure: string, domainStr: StringDomainArgs): FPayload {
+    const domain = toDomainArgs(domainStr);
+    const dataEntries = this.filterDataEntriesFromDomain(this.dataEntries, domain);
+    if (dataEntries.length === 0) {
+      return { value: "" };
+    }
+    const values = dataEntries
+      .map((value) => value[measure])
+      .filter((cell) => cell && cell.type !== CellValueType.empty);
+    const aggregator = this.getMeasure(measure).aggregator || "count";
+    const operator = AGGREGATORS_FN[aggregator];
+    if (!operator) {
+      throw new EvaluationError(`Aggregator ${aggregator} does not exist`);
```

Comment:
Either it's a user-facing error, then the message should be translated and `aggregator` correctly interpolated with %s, or it should be `Error` instead of `EvaluationError`